### ISSUE TITLE
[RL] Dynamic Batching in RL loop

### ISSUE THREAD

**Issue by wwwjn** [pytorch/torchtitan#2693]
Now the trainer takes a single episode, don't do any padding or packing.

Need to implement Dynamic batching for training efficiency: 

<img width="1704" height="892" alt="Image" src="https://github.com/user-attachments/assets/a06258ef-7550-4fce-9363-53b5eb21ddba" />

Related discussion:
1. https://github.com/pytorch/torchtitan/pull/2568#discussion_r2970611002
